scipy/stats/_distn_infrastructure.py

```diff
@@ -1056,10 +1057,12 @@ def stats(self, *args, **kwds):
                     mu2p = self._munp(2, *goodargs)
                     if mu is None:
                         mu = self._munp(1, *goodargs)
-                    mu2 = mu2p - mu * mu
-                    if np.isinf(mu):
-                        # if mean is inf then var is also inf
-                        mu2 = np.inf
+                    # if mean is inf then var is also inf
+                    mu2 = _lazyselect(
+                        [np.isfinite(mu)],
+                        [lambda mu: mu2p - mu**2],
+                        [mu],
+                        default=np.inf)
```

Comment:
Is this equivalent to:
```
mu2 = np.where(np.isfinite(mu), mu2p - mu * mu, np.inf)
```
?

Reply:
Agreed, `where` is more appropriate than `select` in this case.

`mu2p - mu**2` can give ∞ - ∞ NaN results, so you need to protect the `where` expression using
```python
with np.errstate(invalid='ignore'):
    mu2 = np.where(np.isfinite(mu), mu2p - mu**2, np.inf)
```

Alternatively, using `_lazywhere` this is:
```python
mu2 = _lazywhere(np.isfinite(mu), (mu,), lambda mu: mu2p - mu**2, fillvalue=np.inf)
```

I guess the former is more readable.

Regarding `mu**2` vs,. `mu*mu`, the two run at approximately the same speed on my tests so no reason to prefer one over the other.